_stbt/core.py

```diff
@@ -490,6 +490,19 @@ def __repr__(self):
                 "<Custom Image>" if isinstance(self.image, numpy.ndarray)
                 else repr(self.image)))
 
+    def __nonzero__(self):
+        return self.match
+
+    def __eq__(self, other):
+        return (self.match == other.match and self.region == other.region and
+                self.image == other.image)
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+    def __hash__(self):
+        return hash((self.match, self.region, self.image))
```

Comment:
I have to admit, this makes me a little nervous.  I can't think of any particular scenario that's someone's likely to do in practice where it would go wrong but it would certainly be a surprising if you were to discover these semantics accidentally.

We're a bit cheeky and override equality in `FrameObject`s too, ignoring the `_frame` and other protected members of the class even though sometimes we assume their existence, but those members are at least protected.

Maybe it's fine.  These semantics certainly seem useful - more useful in practice than including the `first_pass_result`.  Maybe the real mistake was making `frame` a member at all, but we're unlikely to do anything about that at this point.